src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -136,13 +136,25 @@ private static OperationalStatus GetOperationalStatus(string name)
         // Maps values from /sys/class/net/<interface>/operstate to OperationStatus values.
         private static OperationalStatus MapState(string state)
         {
-            // TODO (#7889): Figure out the possible values that Linux might return.
+            //
+            // http://users.sosdg.org/~qiyong/lxr/source/Documentation/networking/operstates.txt?a=um#L41
+            //
             switch (state)
             {
-                case "up":
-                    return OperationalStatus.Up;
+                case "unknown":
+                    return OperationalStatus.Unknown;
+                case "notpresent":
+                    return OperationalStatus.NotPresent;
                 case "down":
                     return OperationalStatus.Down;
+                case "lowerlayerdown":
+                    return OperationalStatus.LowerLayerDown;
+                case "testing":
+                    return OperationalStatus.Testing;
+                case "dormant":
+                    return OperationalStatus.Dormant;
+                case "up":
+                    return OperationalStatus.Up;
```

Comment:
That sounds like a Functional test (since you need to modify global machine state). 
What I mean by Unit Test is to pull in enough of the sockets product code and fake system calls. The Fake class would decouple the code from the system layer and provide you with full control over how it behaves.

We already have other UnitTests for NetworkInformation: https://github.com/dotnet/corefx/blob/master/src/System.Net.NetworkInformation/tests/UnitTests/NetworkInterfaceTest.cs